rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/Sw360ComponentService.java

```diff
@@ -168,6 +170,42 @@ public List<Component> searchComponentByName(String name) throws TException {
         return sw360ComponentClient.searchComponentForExport(name.toLowerCase(), false);
     }
 
+    public List<Release> getReleasesByComponentId(String id,User user) throws TException {
+        ComponentService.Iface sw360ComponentClient = getThriftComponentClient();
+        return sw360ComponentClient.getReleaseByComponentId(id, user);
+    }
+
+    public List<ReleaseLink> convertReleaseToReleaseLink(String id,User user) throws TException {
+        List<Release> releases = getReleasesByComponentId(id,user);
+        List<ReleaseLink> releaseLinks = new ArrayList<>();
+        releases.forEach(release -> {
+            ReleaseLink releaseLink =new ReleaseLink();
+            releaseLink.setId(release.getId());
+            releaseLink.setName(release.getName());
+            releaseLink.setVersion(release.getVersion());
+            releaseLink.setClearingState(release.getClearingState());
+            boolean checkTypeAttachment = checkAttachmentType(release);
+            if (checkTypeAttachment) {
+                releaseLink.setClearingReport("no report");
+            } else {
+                releaseLink.setClearingReport("");
+            }
+            releaseLink.setMainlineState(release.getMainlineState());
+            releaseLinks.add(releaseLink);
+        });
+
+        return releaseLinks;
+    }
+
+    private Boolean checkAttachmentType(Release release){
```

Comment:
This method can be simplified using `anyMatch` instead of `filter` and `count`


```private boolean checkAttachmentType(final Release release) {
    final Set<Attachment> attachments = release.getAttachments();
    return attachments.stream()
            .anyMatch(attachment -> attachment.getAttachmentType().equals(AttachmentType.COMPONENT_LICENSE_INFO_XML)
                    || attachment.getAttachmentType().equals(AttachmentType.CLEARING_REPORT));
}